packages/integration-sdk-runtime/src/synchronization/batchBySize.ts

```diff
@@ -0,0 +1,102 @@
+import {
+  Entity,
+  IntegrationLogger,
+  IntegrationWarnEventName,
+  Relationship,
+} from '@jupiterone/integration-sdk-core';
+import { UploadDataLookup } from '.';
+import { MAX_BATCH_SIZE_IN_BYTES } from './shrinkBatchRawData';
+
+/**
+Converts an array of graphObjects into an array of groups of this objects.
+@param graphObjects  Entities or relationships to group.
+
+@param maximumBatchSizeInBytes The maximum size of the group of objects.
+
+@param logger 
+ */
+export function batchGraphObjectsBySizeInBytes<
+  T extends UploadDataLookup,
+  K extends keyof T,
+>(
+  graphObjects: T[K][],
+  maximumBatchSizeInBytes: number,
+  logger: IntegrationLogger,
+): T[K][][] {
+  const batches: T[K][][] = [];
+
+  let currentBatch: T[K][] = [];
+  let currentBatchSize: number = 0;
+  for (const graphObject of graphObjects) {
+    const typedGraphObject = graphObject as Entity | Relationship;
+
+    let graphObjectSizeInBytes = getSizeOfObject(graphObject);
```

Comment:
I think this needs to be try/caught - in the event that `getSizeOfObject()` throws, we also need to alert the user that we cannot safely upload this object.

```suggestion
    let graphObjectSizeInBytes = 0;
    try { 
      graphObjectSizeInBytes = getSizeOfObject(graphObject);
    } catch (err) {
      logger.warn({ err }, 'Failed to calculate size of object in bytes');
      logger.publishWarnEvent({ ... });
      continue;
    }
```